python/cuml/cuml/common/sparse_utils.py

```diff
@@ -35,8 +31,7 @@ def is_sparse(X):
     is_sparse : boolean
         is the input sparse?
     """
-    is_scipy_sparse = has_scipy() and scipy.sparse.issparse(X)
-    return cupyx.scipy.sparse.issparse(X) or is_scipy_sparse
+    return scipy.sparse.issparse(X) or cupyx.scipy.sparse.issparse(X)
```

Comment:
Do we have to worry about the order in which the parts of the `or` statement are evaluated? I think Python short circuits logical statements like this, aka it stops once the value is known. I can't work out in my head if the old version relied on this behaviour to avoid errors? Maybe also far too obscure a thing to worry about?

Reply:
Maybe https://github.com/rapidsai/cuml/pull/6596/files#diff-c3c4164cf77550054ea884f2df9f240020f66dcdb29e69e3eabedb308f5fd246R224 is the answer: too obscure to worry about